evm/src/main/java/org/hyperledger/besu/evm/Code.java

```diff
@@ -28,21 +32,38 @@
   /** The bytes representing the code. */
   private final Bytes bytes;
 
+  /** The hash of the code, needed for accessing metadata about the bytecode */
+  private final Hash codeHash;
+
   /** Used to cache valid jump destinations. */
   //  private BitSet validJumpDestinations;
   long[] validJumpDestinations;
 
+  /** Syntactic sugar for an empty contract */
+  public static Code EMPTY = new Code(Bytes.EMPTY, Hash.EMPTY);
+
+  private static final Logger LOG = getLogger();
+
   /**
    * Public constructor.
    *
    * @param bytes The byte representation of the code.
+   * @param codeHash the Hash of the bytes in the code.
    */
-  public Code(final Bytes bytes) {
+  public Code(final Bytes bytes, final Hash codeHash) {
     this.bytes = bytes;
+    this.codeHash = codeHash;
+  }
+
+  public Code(final Bytes bytecode, final Hash codeHash, final long[] validJumpDestinations) {
```

Comment:
Static side effects are not appropriate for constructors.  If you need side effects do it in a factory method that calls the constructor and then does the side effects.